com.unity.shadergraph/Editor/Drawing/MaterialGraphEditWindow.cs

```diff
@@ -721,14 +738,18 @@ public void ToSubGraph()
 
             // Collect the keyword nodes and get the corresponding keywords
             var keywordNodes = graphView.selection.OfType<IShaderNodeView>().Where(x => (x.node is KeywordNode)).Select(x => ((KeywordNode)x.node).keyword);
+            var dropdownNodes = graphView.selection.OfType<IShaderNodeView>().Where(x => (x.node is DropdownNode)).Select(x => ((DropdownNode)x.node).dropdown);
```

Comment:
Given that dropdowns can only ever be present in a sub graph, could we skip this statement if we're in a parent graph? could save a little bit of execution time on the LINQ expression

Reply:
Might be a premature optimization if the branch costs more than the search expression itself, so might be better to profile it before saying one way or the other

Reply:
I think this is inside the call to "ToSubGraph". How does that get invoked? I could not find a call to it

Reply:
If you select a group of nodes, right click and then in the "Convert To" menu, you'll find the "To SubGraph" option

Reply:
Got it, I see now. IMO it makes most sense to leave this generic, and use the logic that already exists to allow/prevent invalid nodes being pasted into the corresponding target graph. I don't think there's value in adding extra complexity to the code here. It's a one time action, and I don't see any performance motivation for doing this.